docs/details/develop-checks.md

```diff
@@ -1,266 +1,313 @@
-
-Checks format and the current STT UI use different terminology for severities. Here is how different formats show up on the UI:
-
-| Format    | UI       |
-| --------- | -------- |
-| emergency |          |
-| alert     |          |
-| critical  |          |
-| error     | Critical |
-| warning   | Major    |
-| notice    | Trivial  |
-| info      |          |
-| debug     |          |
-
+  
 ## Backend
 
-![!](../_images/BackendSTT.png)
-
-1. pmm-managed checks that this installation is opted-in into STT.
-2. pmm-managed downloads checks file from SaaS.
+1. pmm-managed checks that the installation is opted-in for checks.
+2. pmm-managed downloads files with checks from SaaS.
 3. pmm-managed verifies file signatures using a list of hard-coded public keys. At least one signature should be correct.
 4. pmm-managed sends queries to pmm-agent and gathers results.
 5. pmm-managed executes check scripts that produce alert information.
-6. pmm-managed sends alerts to Alertmanager.
-   - Due to Alertmanager design, pmm-managed has to send and re-send alerts to it much more often than the frequency of check execution. That's expected, and not important for using STT, but important for understanding how it works.
+6. pmm-managed sends alerts to Alermanager.
+   - Due to Alertmanager design, pmm-managed has to send and re-send alerts to it much more often than the frequency with which checks are executed. This expected behaviour is not important for using checks but is important for understanding how checks work.
    - Currently, Prometheus is not involved.
 
-## Frontend
-
-![!](../_images/FrontEndSTT.png)
+![!](../_images/BackendChecks.png)
 
-Our UI in Grafana uses Alertmanager API v2 to get information about failed security checks.
+## Frontend
+PMM UI in Grafana uses Alermanager API v2 to get information about failed checks:
 
+![!](../_images/FrontEndChecks.png)
+ 
+ 
+## Check fields
+Checks include the following fields:  
+
+- **Version** (integer, required): defines what other properties are expected, what types are supported, what is expected from the script and what it can expect from the execution environment, etc.
+- **Name** (string, required): defines machine-readable name (ID).
+- **Summary** (string, required): defines short human-readable description.
+- **Description** (string, required): defines long human-readable description.
+- **Type** (string/enum, required): defines the query type and the PMM Service type for which the advisor runs. Check the list of available types in the table below.
+- **Query** (string, optional): contains an SQL query as a string with proper quoting. 
+    The query is executed on the PMM Client side and can be absent if the type defines the whole query by itself.  
+- **Script** (string, required): contains a small Python program that processes query results, and returns check results. It is executed on the PMM Server side.
```

Comment:
I think better to use `Starlark` instead of `Python` everywhere, it's more correct.